File: fixtures/m24/rust/src/main.rs
mod greeting {
    pub trait Greeter {
        fn greet(&self, name: &str) -> String;
    }

    pub struct EnglishGreeter;

    impl Greeter for EnglishGreeter {
        fn greet(&self, name: &str) -> String {
            format!("Hello {name}")
        }
    }
}

use greeting::{EnglishGreeter, Greeter};

fn main() {
    let greeter = EnglishGreeter;
    println!("{}", greeter.greet("MINOS"));
}

File: fixtures/m24/go/main.go
package main

import "fmt"

type Greeter interface {
	Greet(name string) string
}

type englishGreeter struct{}

func (englishGreeter) Greet(name string) string {
	return "Hello " + name
}

func main() {
	var greeter Greeter = englishGreeter{}
	fmt.Println(greeter.Greet("MINOS"))
}
